src/chain.h

```diff
@@ -303,10 +303,10 @@ class CBlockIndex
 
     std::string ToString() const
     {
-        return strprintf("CBlockIndex(pprev=%p, nHeight=%d, merkle=%s, hashBlock=%s)",
+        return strprintf("CBlockIndex(pprev=%p, nHeight=%d, merkle=%s%s)",
             pprev, nHeight,
             hashMerkleRoot.ToString(),
-            GetBlockHash().ToString());
+            phashBlock ? ", hashBlock=" + phashBlock->ToString() : "");
```

Comment:
0ddad240dcd43b46ee70ccfb112d66d655b5c34f:

Missing the hash is a bug. I think it is better to assert than to silently continue when there is a bug.

Reply:
Good idea, updating.